watchmal/engine/engine_classifier.py

```diff
@@ -384,7 +391,10 @@ def evaluate(self, test_config):
                 predictions.extend(result['predicted_labels'])
                 softmaxes.extend(result["softmax"])
 
-                print("eval_iteration : " + str(it) + " eval_loss : " + str(result["loss"]) + " eval_accuracy : " + str(result["accuracy"]))
+                if report_test_metrics:
+                    print("eval_iteration : " + str(it) + " eval_loss : " + str(result["loss"]) + " eval_accuracy : " + str(result["accuracy"]))
+                else:
+                    print("eval_iteration : " + str(it))
 
                 eval_iterations += 1
         
```

Comment:
The loss/accuracy is used in a few places below, have you checked this doesn't cause problems? Specifically:
- Lines 405-406 converts them to numpy arrays (should be fine, just empty arrays)
- Line 419 synchronises them when using DDP (what happens if they're empty arrays?)
- Line 424 converts them from torch tensors to numpy arrays (probably still fine if they're still empty tensors)
- Lines 442-448 prints them out (it'll just print zeros, but maybe better to skip this based on the config?)
 
Not sure if it's better to add a load of if statements to skip this when metrics are disabled, or to avoid adding too many if statements where it doesn't matter that it's just processing empty lists/arrays/tensors.

Reply:
I've checked and these all seem to be fine, the cumulative loss is just 0 and gets printed out as such, and the synchronization steps just operate on empty arrays

Reply:
Okay, sounds fine then.